### PR TITLE
Cleaner, better gooso targetting of item drop based monsters

### PR DESCRIPTION
**Pull request by Ignose**

### CODE REVIEW

File: packages/garbo/src/familiar/itemFamiliar.ts [hunk 1]
```diff
@@ -68,6 +71,36 @@ export function bestFairy(): Familiar {
...
+      {
+        familiar: $familiar`Grey Goose`,
+        expectedValue:
+          !get("_meatifyMatterUsed") &&
+          $familiar`Pocket Professor`.experience < 400
+            ? getModifier("Familiar Experience") * (15 ** 4 / 400)
+            : 0, // Meatify Matter is worth 15 ** 4, each xp is worth 1/400th of that
+        leprechaunMultiplier: 0,
+        limit: "none",
+      },
```
Comment (horrible-little-slime) [inline packages/garbo/src/familiar/itemFamiliar.ts:103]: Make this a separate PR, and also just flip the "include experience familiars" switch instead of duplicating code

File: packages/garbo/src/config.ts [hunk 2]
```diff
@@ -98,6 +99,14 @@ function defaultTarget() {
 export const targettingMeat = () =>
   $monsters`cheerless mime executive`.includes(globalOptions.target);
 
+export const targettingItems = () =>
+  valueDrops(globalOptions.target) > meatDrop(globalOptions.target);
+
+export const goosoDroneEligible = () =>
+  targettingItems() &&
+  itemDropsArray(globalOptions.target).length === 1 &&
+  have($familiar`Grey Goose`);
```
Comment (horrible-little-slime) [inline packages/garbo/src/config.ts:108]: Export this from `lib`, and use the word `goose` instead of `gooso` everywhere possible in this PR

**Review by horrible-little-slime — DISMISSED** [review]

File: packages/garbo/src/config.ts [hunk 3]
```diff
@@ -98,6 +99,14 @@ function defaultTarget() {
 export const targettingMeat = () =>
   $monsters`cheerless mime executive`.includes(globalOptions.target);
 
+export const targettingItems = () =>
+  valueDrops(globalOptions.target) > meatDrop(globalOptions.target);
```
Comment (horrible-little-slime) [inline packages/garbo/src/config.ts:103]: Honestly we should probably export these from lib as well, I recognize I'm the one to have added `targettingMeat` but while we're cleaning it up we might as well

[the remark this answers is not in the document]
Comment (horrible-little-slime) [reply packages/garbo/src/config.ts:103]: Leave it as-is for now for sure

File: packages/garbo/src/config.ts [hunk 4]
```diff
@@ -98,6 +99,14 @@ function defaultTarget() {
...
+
+export const goosoDroneEligible = () =>
+  targettingItems() &&
+  itemDropsArray(globalOptions.target).length === 1 &&
```
[the remark this answers is not in the document]
Comment (Ignose) [reply packages/garbo/src/config.ts:107]: How would you do that? Something like dropping all those records out of the itemDropsArray, THEN checking the length?

File: packages/garbo/src/fights.ts [hunk 5]
```diff
@@ -439,6 +439,13 @@ function familiarSpec(underwater: boolean, fight: string): OutfitSpec {
...
+      if (goosoDroneEligible()) {
+        if (get("gooseDronesRemaining") < copyTargetCount()) {
+          return { familiar: $familiar`Grey Goose` };
+        } else {
+          return { familiar: freeFightFamiliar() };
+        }
+      }
     }
   }
   return { familiar: meatFamiliar() };
```
[the remark this answers is not in the document]
Comment (Ignose) [reply packages/garbo/src/fights.ts:465]: I did as requested; but I also added a free fight check to not use meatFamiliar. If it's a free fight, meatFamiliar is a waste of a fight.

**Review by Rinn — APPROVED** [review]
